synapse/api/constants.py

```diff
@@ -70,6 +70,11 @@ class LoginType:
     DUMMY = "m.login.dummy"
 
 
+# This is not a login type as we cannot login with it, however
+# the spec says that this is a valid registration type for appservices.
```

Comment:
a "registration type" isn't really a concept the spec has.

(and no, it's certainly not a login type).

```suggestion
# This is used in the `type` parameter for /register when called by
# an appservice to register a new user.
```

Reply:
Incidentally, the uses of a magic `type: magic_value` parameter with usage completely at odds with the *normal* use of `/register` is a *terrible* idea, but that's a matter for the spec, not here.

(it should be a separate endpoint imho)

Reply:
Agreed, we should have an appservice registration endpoint given they don't even use all the parameters from the response anyway (e.g. they don't care about devices) so perhaps there should be a proposal made for that...